### PR TITLE
Correct CI cache errors and gem release process

### PR DESCRIPTION
**Pull request by invke**
### 📝 Description

Set up building and deploying the gem using `gem-release` which is needed for the CI workflow.

Some other bits:
- Source the gemfile from the gemspec
- Add missing dev deps

This works, tested by just forcing the tag onto this branch so I can stop making PRs to test (why I did that initially I'm not sure but here we are).

See:
- https://github.com/paperkite/paperkite-rubocop/actions/runs/12129083004

### 🌳 Related PRs

- #19 
- #18 
- #17

### CODE REVIEW

File: .github/workflows/rubygems-deployment.yaml [hunk 1]
```diff
@@ -15,4 +15,7 @@ jobs:
     steps:
     - uses: actions/checkout@v4
     - uses: ruby/setup-ruby@v1
+      with:
+        bundler-cache: true
+        ruby-version: 3.3
```
Comment (chillioil) [inline .github/workflows/rubygems-deployment.yaml:20]: This doesn't need the patch verion in this? 3.3.6

as per your pr [here](https://github.com/paperkite/paperkite-rubocop/pull/17/files)

**Review by chillioil — APPROVED** [review]
LGTM, just left a comment.

**Review by LuyiPK — APPROVED** [review]
LGTM

File: .github/workflows/rubygems-deployment.yaml (hunk 1 above)
Comment (invke) [reply .github/workflows/rubygems-deployment.yaml:20]: Nah for deployment it doesn't matter, it'll use latest. It should be detecting it so I probably can remove this line.

**Pull request merged**